### DISCUSSION TITLE
How to get some pretty printing in a string

### DISCUSSION THREAD

**Discussion by beemibrahim** [nlohmann/json#3324] [Q&A] [answered]
I want to dump json contents into a string , then print it with pretty printing

**Comment by nlohmann** [comment, marked as answer]
https://json.nlohmann.me/api/basic_json/dump/

**Comment by beemibrahim** [comment]
Also "dump()" returns a C++ string where the entries are in alphabetical order, i want them to be in normal order

**Reply by gregmarr** [reply]
That's not `dump()`, that's the normal behavior of the `json` type.  There is a type for preserving object order.
https://json.nlohmann.me/features/object_order/

**Comment by lix19937** [comment]
```cpp  
      nlohmann::json js = nlohmann::json::parse(model_desc);
      std::string pretty_s = js.dump(4);
      std::cout << pretty_s << std::endl;
```
